src/components/Registration/Registration.jsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-alert */
+import { Link, useNavigate } from 'react-router-dom';
+import { useState } from 'react/cjs/react.development';
+
+import Button from '../../common/Button/Button';
+import Input from '../../common/Input/Input';
+import { inputText, buttonText, serverAddress } from '../../constants';
+
+import './Registration.css';
+
+async function performRegistration(userName, userEmail, userPassword) {
+  const registrationData = {
+    name: userName,
+    email: userEmail,
+    password: userPassword,
+  };
+
+  const response = await fetch(`${serverAddress}/register`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(registrationData),
+  });
+  const responseJSON = await response.json();
+  return responseJSON;
+}
+
+export default function Registration() {
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [userPassword, setUserPassword] = useState('');
+  const navigate = useNavigate();
```

Comment:
Does this work? I don't see useNavigate in react-router-dom docs. What about useHistory?

Reply:
I used version 6 here of react router dom. And there, useHistory is replaced by useNavigate
https://reactrouter.com/docs/en/v6/upgrading/v5#use-usenavigate-instead-of-usehistory